File: skypydb-js/src/api/operations/vector/collection/update_items.ts
import type { UpdateParams } from "../../../../types";
import { QueryItemsMixin } from "./query_items";

export class UpdateItemsMixin extends QueryItemsMixin {
  async update(params: UpdateParams): Promise<void> {
    const { ids, embeddings, documents, metadatas } = params;
    await this._db.update(this._name, ids, embeddings, documents, metadatas);
  }
}

File: skypydb/api/operations/vector/collection/update_items.py
"""Mixin that updates records in a vector collection."""

from typing import Any, Dict, List, Optional


class UpdateItemsMixin:
    """Expose update behavior on a `Collection` wrapper."""

    def update(
        self,
        ids: List[str],
        embeddings: Optional[List[List[float]]] = None,
        documents: Optional[List[str]] = None,
        metadatas: Optional[List[Dict[str, Any]]] = None,
    ) -> None:
        """Update existing items by ID."""

        self._db.update(
            collection_name=self._name,
            ids=ids,
            embeddings=embeddings,
            documents=documents,
            metadatas=metadatas,
        )
